base_layer/core/src/transaction_protocol/recipient.rs

```diff
@@ -114,9 +113,9 @@ impl ReceiverTransactionProtocol {
     }
 
     /// Retrieve the final signature data to be returned to the sender to complete the transaction.
-    pub fn get_signed_data(&self) -> Result<&RecipientSignedTransactionData, TransactionProtocolError> {
+    pub fn get_signed_data(&self) -> Result<RecipientSignedTransactionData, TransactionProtocolError> {
         match &self.state {
-            RecipientState::Finalized(data) => Ok(&data),
+            RecipientState::Finalized(data) => Ok(data.clone()),
```

Comment:
Can I suggest you leave this as is, and call clone on the client side if you want a copy of the data?

This change makes it expensive to simply query values of the final tx data.